frontend/packages/knative-plugin/src/components/traffic-splitting/TrafficSplittingController.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { K8sResourceKind, K8sKind } from '@console/internal/module/k8s';
+import { Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { createModalLauncher } from '@console/internal/components/factory';
+import {
+  transformTrafficSplitingData,
+  knativeServingResourcesTrafficSplitting,
+} from '../../utils/traffic-splitting-utils';
+import TrafficSplitting from './TrafficSplitting';
+
+type ControllerProps = {
+  loaded?: boolean;
+  obj: K8sResourceKind;
+  resources?: {
+    configurations: FirehoseResult;
+    revisions: FirehoseResult;
+    ksservies: FirehoseResult;
+  };
+};
+
+const Controller: React.FC<ControllerProps> = (props) => {
+  const { loaded, obj, resources } = props;
+  const { service, revisions } = transformTrafficSplitingData(obj, resources);
+  return loaded ? <TrafficSplitting {...props} service={service} revisions={revisions} /> : null;
+};
+
+type TrafficSplittingControllerProps = {
+  obj: K8sResourceKind;
+  kind: K8sKind;
+};
+
+const TrafficSplittingController: React.FC<TrafficSplittingControllerProps> = (props) => {
+  const {
+    metadata: { namespace },
+  } = props.obj;
+  const resources = knativeServingResourcesTrafficSplitting(namespace);
+
+  return (
+    <Firehose resources={resources}>
+      <Controller {...props} />
+    </Firehose>
+  );
+};
+
+export const trafficModalLauncher = createModalLauncher<any>(TrafficSplittingController);
```

Comment:
Instead of `any` you can try this:
`type Props = TrafficSplittingControllerProps & ModalComponentProps;`
`export const trafficModalLauncher = createModalLauncher<Props>(TrafficSplittingController);`
Imprort the `ModalComponentProps` from `@console/internal/components/factory`